### DISCUSSION TITLE
Add inkeep or customer support chatbot equivalent to help people on discord

### DISCUSSION THREAD

**Discussion by waynehamadi** [twentyhq/twenty#5698] [Ideas]
Would be game changer and would reduce your churn.
I am sure people get stuck on the same problem and then stop or move on, at least having this thing would allow them to have some sort of relief.

I can tell you on windmill's discord, inkeep is working very well.

**Comment by greptile-apps** [comment]
**Disclaimer:** This comment was automatically generated with AI. It can give some useful hints as to where to look for, but sometimes it can also give very inaccurate answers so takes this with a pinch of salt :).


To add a customer support chatbot to Discord, follow these steps:

1. **Review Zapier Integration Documentation**: Refer to `/packages/twenty-docs/docs/contributor/server/zapier.mdx` for guidance on setting up integrations.

2. **API Key Management**: Use the existing framework for API key management found in `/packages/twenty-front/src/pages/settings/developers/SettingsDevelopers.tsx`.

3. **Create New Integration**: Implement a new section for the chatbot integration similar to the existing API keys and webhooks sections.

4. **Discord Bot Setup**: Follow the Discord developer documentation to create and configure a bot. Use the API keys generated in the previous step to authenticate and manage the bot.

5. **Deploy and Test**: Deploy the bot and test its functionality within your Discord server.



### References

[/packages/twenty-docs/docs/contributor/bug-and-requests.mdx](https://github.com/twentyhq/twenty/blob/main/packages/twenty-docs/docs/contributor/bug-and-requests.mdx)
[/packages/twenty-docs/docs/contributor/server/zapier.mdx](https://github.com/twentyhq/twenty/blob/main/packages/twenty-docs/docs/contributor/server/zapier.mdx)
[/packages/twenty-front/src/pages/settings/developers/SettingsDevelopers.tsx](https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/pages/settings/developers/SettingsDevelopers.tsx)
[/packages/twenty-front/src/pages/settings/developers](https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/pages/settings/developers)


[**Ask Greptile**](https://app.greptile.com/chat/github/twentyhq/twenty/main)

**Comment by FelixMalfait** [comment]
I'm afraid we don't have enough documentation yet for something like Inkeep (unlike windmill). I'll convert this to a discussion to come back in a few months.
 
 Maybe Greptile can also do it at somepoint (uses code not docs for now), that way we'd have a single AI assistant we can enrich that is on issues/prs/discord.
 You can check their codebase chat https://app.greptile.com/chat (replace twentyhq/twenty in the url after picking a popular repo)

**Comment by BOHEUS** [comment]
Now there's Twenty bot on Discord server answering users questions on #help-ai
